src/inc/sift-events/sift-events.php

```diff
@@ -851,8 +898,23 @@ public static function send() {
 			}
 
 			foreach ( self::$to_send as $entry ) {
+				// We need the original WC user ID to handle the decision locally.
+				$wccom_user_id = $entry['properties']['$user_id'] ?? null;
+
+				// Apply the filter to update from the WCCOM to the WPCOM user_id just before sending the event.
+				$entry['properties'] = apply_filters( 'sift_for_woocommerce_pre_send_event_properties', $entry['properties'] );
```

Comment:
`sift_for_woocommerce_pre_send_event_properties` should not be called here. It is meant to allow properties modifications before sending to Sift. What you need here would be the reverse operation. See [this comment](https://github.com/woocommerce/sift-for-woocommerce/pull/24/files#r1860998263)